docs/developers-guide/build.md

```diff
@@ -1,25 +1,132 @@
 
 ```
 sudo apt install openjdk-11-jdk nodejs && sudo npm install --global yarn
 ```
-If you have multiple JDK versions installed in your machine, be sure to switch your JDK before building by doing `sudo update-alternatives --config java` and selecting Java 11 in the menu
+
+If you have multiple JDK versions installed in your machine, be sure to switch your JDK before building with: 
+
+```
+sudo update-alternatives --config java
+``` 
+
+Then select Java 11 in the menu.
 
 If you are developing on Windows, make sure to use Ubuntu on Windows and follow instructions for Ubuntu/Linux instead of installing ordinary Windows versions.
 
 Alternatively, without the need to explicitly install the above dependencies, follow the guide [on using Visual Studio Code](deven.md#developing-with-visual-studio-code.md) and its remote container support.
 
+## Copy the Metabase code to your computer
+
+Once you have all that installed, you’re going to get a copy of Metabase's source code from GitHub, where it’s stored.
+
+1. Create a `workspace` folder (you can name it that or whatever you want). You’ll put the Metabase code files inside this `workspace` folder.
+
+2. Open up your terminal app, and navigate to your workspace folder with: 
+
+```
+cd ~/workspace
+```
+
+1. Run this GitHub command to “clone” Metabase into this folder
+
+```
+gh repo clone metabase/metabase
+```
+
+## Choose the branch you want to run, and run it!
+
+This is the part that you’ll use over and over. 
+
+The “official” branch of Metabase is called `master`, and other feature development branches get merged into it when they’re approved. So if you want to try out a feature before then, you’ll need to know the name of that branch so you can switch over to it. Here’s what to do:
+
+1. Open up your terminal app
+
+2. Navigate to where you're storing the Metabase code. If you followed this guide exactly, you'd get there by entering this command: 
+   
+   ```
+   cd ~/workspace/metabase
+   ```
+
+3. "Pull” down the latest code by running: 
+
+   ```
+   git pull
+   ```
+
+   You should do this every time to make sure that you have all the latest Metabase branches and code on your computer. It’s also how you’ll get updates on a feature branch if the engineer make changes to it.
+
+4. Find the name of the branch you want to run by going to the “pull request” page for that feature on GitHub and copying the branch name from there. Here’s [an example PR page](https://github.com/metabase/metabase/pull/19138), with the branch name
+`fix-native-dataset-drill-popover`.
+
+5. Switch to, or “check out,” that branch by running:
+
+   ```
+   git checkout branch-name
+   ```
+    
+   If we wanted to switch to the branch in the previous step, we'd run:
+
+   ```
+   git checkout fix-native-dataset-drill-popover
+   ```
+    
+   When you want to switch back to `master`, run: 
+    
+   ```
+   git checkout master
+   ```
+
+6. Now we’ll start up the backend server of Metabase with:
```

Comment:
This should be its own section, as it talks about running the code (not getting the code).